### PR TITLE
LOG-4603: Add log forwarder input selectors for content filtering

### PR DESCRIPTION
**Pull request by jcantrill**
This enhancement doc:

* Adds log forwarder API to spec log input selection based upon kubernetes metadata

cc @alanconway @cahartma @xperimental @periklis

### CODE REVIEW

**Comment by r2d2rnd** [comment]
Hello @jcantrill, 

From now, I believe in the goals section should be good to add after:

* Reduce the CPU and memory load on the collector by configuring it to only process logs that are interesting to administrators

These other two points:
* Reduce the network usage when log forwarding
* Reduce the storage required on the log store side

File: enhancements/cluster-logging/forwarder-input-selectors.md [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+
+
+* As an administrator of cluster logging, I want to only forward logs from a limited set of namespaces because I do not need the others
+* As an administrator of cluster logging, I want to exclude logs from a limited set of namespaces because I do not need them
+* As an administrator of cluster logging, I want to only forward logs from pods with a specific set of labels
+* As an administrator of cluster logging, I want to exclude certain container logs from a pod because they are noisy and uninteresting to me
+* As an administrator of cluster logging, I do not want to collect node logs because they are not of interest to me
+
+
+### Goals
```
Comment (r2d2rnd) [inline enhancements/cluster-logging/forwarder-input-selectors.md:86]: I'd add these other two points:

*  Reduce the network usage when log forwarding
*  Reduce the resources required on the log store side related to storage size, cpu and memory

**Comment by bysnupy** [comment]
It looks good to me. Thanks.

**Comment by alanconway** [comment]
+1 see related filtering enhancement https://github.com/openshift/enhancements/pull/1513

**Review by alanconway — APPROVED** [review]
/lgtm

**Pull request merged**